coap/api/transport.go

```diff
@@ -112,14 +116,30 @@ func authorize(msg *gocoap.Message, res *gocoap.Message, cid string) (string, er
 	return id.GetValue(), nil
 }
 
-func fmtSubtopic(subtopic string) string {
-	if subtopic != "" {
-		if strings.HasSuffix(subtopic, "/") {
-			subtopic = subtopic[:len(subtopic)-1]
+func fmtSubtopic(subtopic string) (string, error) {
+	if subtopic == "" {
+		return subtopic, nil
+	}
+
+	subtopic = strings.Replace(subtopic, "/", ".", -1)
+
+	elems := strings.Split(subtopic, ".")
+	filteredElems := []string{}
+	for _, elem := range elems {
+		if elem == "" {
+			continue
+		}
+
+		if len(elem) > 1 && (strings.Contains(elem, "*") || strings.Contains(elem, ">")) {
```

Comment:
Why do we exclude joker topics?

Reply:
We didn't, we just exclude invalid ones, like `fo*o`.